### ISSUE TITLE
cappingMinimumTimeInterval seems to be not working for me

### ISSUE THREAD

**Issue by Niyatsu** [dailymotion/vast-client-js#362]
Hey, I'm creating an instance of VASTClient with a cappingMinimumTimeInterval following the documentation, just like this: 
```javascript
const vastClient = new VASTClient(0, 900000);
````
but it seems to ignore the interval, and I get consecutive requests.
And I run is just like this: 
```javascript
playtBtn.addEventListener("click", () => {
  mainPlayer.addEventListener("play", requestAds);
});

function requestAds() {
  vastClient
    .get(dviadlink)
    .then((res) => {...

````
After running an ad (successful request) I hit the play button and get another request.
Am I doing something wrong, I'm not sure? For now the property just seems to be ignored somehow.

The library version is 2.5.1. #

**Comment by ZacharieTFR** [comment]
Hello @Niyatsu,

Thanks for creating the issue, indeed I can reproduce it. I will create a PR to fix it 👍 

Just to let you know the most recent version is the pre-release 3.0.0-alpha.10, it's stable and we will release officially the 3.0-version soon.

**Comment by Niyatsu** [comment]
Thanks! I'll try to update asap. :)

**Comment by ZacharieTFR** [comment]
After digging into the code, it turns out that the documentation is missleading. Indeed I was able to reproduce the parameter having no effect because it was not the right way to use it. 
In fact the cappingMinimumTimeInterval is not the time between 2 requests but between 2 ads played. This means that the vast Client has to detect (using VAST Tracker) that the ad has been played. I will update the documentation to make it clearer. 

In order to make it work in your case, you first need to initialize a VAST Tracker with the ad and the creative that you choosed. 
```js
function requestAds() {
  vastClient
    .get(dviadlink)
    .then((res) => {
      const ad = res.ads[0];
      vastTracker = new VASTTracker(vastClient, ad, ad.creatives[0]);
``` 
Then trigger the start tracker when the ad start 
```js
 vastTracker.track('start');
```
And after doing that if you try to execute `vastClient.get()` before the interval it will trigger the error.
This is a short term solution but as a long term solution I think that we could improve it by updating the interval directly when get() call is done and not when the ad has been played. 

Tell me if it solve your issue. If you want to speed up the process for the long term solution feel free to open a PR with the change and it will be reviewed and merged !

**Comment by Niyatsu** [comment]
Thanks, @ZacharieTFR ! Unfortunately, I'm unable to check out the short term solution, as somehow one of my VAST-tags doesn't get parsed. I'm not sure if I should ask for help in a new issue, or whether this repository even a place for such a question, but one of my tags is just always seen empty by VASTClient. Though, when you open the tag in browser it's not empty. (https://ads2.adviad.com/www/delivery/fc.php?script=apVideo:vast2&offset=true&mp4=1&zoneid=787).  
![ads](https://user-images.githubusercontent.com/25269503/86221483-044f0f00-bb96-11ea-9b3c-0a502753baaf.png)
The second response is OK, but the first is from the tag I mentioned and I wonder if it is because it is version 2.0?

**Comment by ZacharieTFR** [comment]
Here, it's fine. Could you provide me the corresponding VAST ? When I follow the link I get an empty VAST: 
```xml
<VAST xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.0" xsi:noNamespaceSchemaLocation="vast.xsd">
<!--  You are requesting vast xml for zone  which does not apear to be a video overlay banner nor a vast inline banner.  -->
</VAST>
``` 
The VAST version shouldn't be a problem, but to benefit from all the lastest feature of the VAST-Client you may want to use the [3.0-version](https://github.com/dailymotion/vast-client-js/tree/3.0-version).

**Comment by Niyatsu** [comment]
@ZacharieTFR Sure, here it is:
`<VAST xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.0" xsi:noNamespaceSchemaLocation="vast.xsd">
<Ad id="pre-roll">
<InLine>
<AdSystem>
<![CDATA[ Adviad ]]>
</AdSystem>
<AdTitle>
<![CDATA[ Adviad-reklam.mp4 ]]>
</AdTitle>
<Description>
<![CDATA[ Inline Video Ad ]]>
</Description>
<Impression id="primaryAdServer">
<![CDATA[ https://ads2.adviad.com/www/delivery/lg.php?bannerid=2027&campaignid=1208&zoneid=787&OACBLOCK=86400&OACCAP=5&cb=a3febb8bd5&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Impression>
<Creatives>
<Creative sequence="1" id="2027">
<Linear skipoffset="00:00:05">
<Duration>00:00:30</Duration>
<VideoClicks>
<ClickThrough>
<![CDATA[ https://ads2.adviad.com/www/delivery/ck.php?oaparams=2__bannerid=2027__zoneid=787__cb=a3febb8bd5__campaignid=1208__p1=1593607505__p2=fae649061d521665efc8551e8655__p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e__oadest=https%3A%2F%2Fadviad.com%3Futm_content%3DAdviad%26utm_source%3Dadviad.com%26utm_medium%3Dnative_video%26utm_campaign%3Dyoda.az_-_Test&button_type=learn_more ]]>
</ClickThrough>
</VideoClicks>
<MediaFiles>
<MediaFile delivery="progressive" bitrate="400" width="640" height="480" type="video/mp4" html="false">
<![CDATA[ https://ads.adviad.com/www/images/11267a95d3f3c237c63a0bda3985e640.mp4 ]]>
</MediaFile>
</MediaFiles>
<TrackingEvents>
<Tracking event="start">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=start&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="midpoint">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=midpoint&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="firstQuartile">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=firstquartile&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="thirdQuartile">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=thirdquartile&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="complete">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=complete&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="mute">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=mute&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="pause">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=pause&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="replay">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=replay&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="fullscreen">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=fullscreen&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="stop">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=stop&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="unmute">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=unmute&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
<Tracking event="resume">
<![CDATA[ https://ads2.adviad.com/www/delivery/fc.php?script=videoAds:vastEvent&bannerid=2027&zoneid=787&event=resume&cb=a3febb8bd5&campaignid=1208&p1=1593607505&p2=fae649061d521665efc8551e8655&p3=364228005.d3446cfba164015a2064e87df46f9480655d1a4e ]]>
</Tracking>
</TrackingEvents>
</Linear>
</Creative>
</Creatives>
</InLine>
</Ad>
</VAST>`

**Comment by ZacharieTFR** [comment]
Thanks 👍  
When I mentionned the 3.0-version it was the vast-client version which is not related to VAST version as the vast-client supports supports VAST versions up to 4.2 and is backward compatible with previous versions (2.0 included).

Using the VAST that you gave me, and the same vast-client version that you're using (2.5.1), I do get a parsed ad as result. In you r example I just replaced the variable `dviadlink` by `'https://run.mocky.io/v3/0fe6ed78-da9e-4cdc-b297-fa0f63478fe7'` (which is your VAST but hosted on a endpoint, please tell me if you want me to delete it) and it work fine. 

In my opinion your issue doesn't come from the vast-client itself but the URL that you use that return an empty VAST. To check it please open the browser developper tools and check in the network tab the response to the call. 
e.g with Chrome and using https://ads2.adviad.com/www/delivery/fc.php?script=apVideo:vast2&offset=true&mp4=1&zoneid=787 as VAST url 
![image](https://user-images.githubusercontent.com/18121007/86248724-8cd9a980-bbae-11ea-9464-690cd27c6647.png)

**Comment by Niyatsu** [comment]
@ZacharieTFR That's true, I checked the mock link you provided and it worked for me as well in my app. Funny thing is, I checked the original link (https://ads2.adviad.com/www/delivery/fc.php?script=apVideo:vast2&offset=true&mp4=1&zoneid=787) in a couple of VAST testers (like this one: http://tools.springserve.com/tagtest) and it works there. That's what confused be in the first place. But I guess I'll try to update vast-client now and then try to communicate with provider.
